src/cg.jl

```diff
@@ -237,3 +240,18 @@ function cg!(x, A, b;
 
     log ? (iterable.x, history) : iterable.x
 end
+
+"""
+    cocg(A, b; kwargs...) -> x, [history]
+
+Same as [`cocg!`](@ref), but allocates a solution vector `x` initialized with zeros.
```

Comment:
I would just have a single docstring for both methods, of the form:
```jl
    cocg!([x=0,] A, b; kwargs...) -> x, [history]

Same as [`cg!`](@ref), ...
```

Reply:
The difference between the two methods is not only `x`, but also the exclamation mark: `cocg!(x, ...)` vs. `cocg(...)`.  Is there a good way to combine the docstrings into one in this case?

Reply:
Whoops, I missed that; you're right, you should have two separate docstrings.